src/app/utils/hooks.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  ADD_USERS_ROUTE,
+  API_URL,
+  FETCH_MEMEBRS_ROUTE,
+  USERS_ROUTE,
+} from "./constants";
+import { User } from "./interfaces";
+
+export const getUsers = async (): Promise<User[]> => {
+  const response = await fetch(`${API_URL}${USERS_ROUTE}`);
+  let data: User[] = [];
+  try {
+    data = await response.json();
+    return data;
+  } catch (error) {
+    console.error(error);
+    return [];
+  }
+};
+
+export const addUsers = async () => {
+  const members = await fetch(`${API_URL}${FETCH_MEMEBRS_ROUTE}`);
+  const users = (await members.json()) as { members: User[] };
+  const res = await fetch(`${API_URL}${ADD_USERS_ROUTE}`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(users),
+  });
+
+  if (res.status === 200) {
+    return true;
+  }
+  return false;
+};
```

Comment:
Instead of this approach, I would recommend having a single API endpoint instead of two. Aka replacing `/fetchactivemembers` and `/addusers` with `/refetch_active_members` or something.

Then inside that endpoint you could
1. Fetch the updated data from members
2. Insert new mongodb documents for the members that are not loaded already

That way you can fetch the members api and all the member data without exposing any of it to the frontend

This would also help you not require validating the member info that is sent from the frontend